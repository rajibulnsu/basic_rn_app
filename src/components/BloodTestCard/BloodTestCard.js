import React from 'react';
import { View, Text, Button } from 'react-native';
import styles from './BloodTestCard.style';

const BloodTestCard = props => {
  const { title, date } = props.item;
  return (
    <View style={styles.cardLayout}>
      <View style={styles.text}>
        <Text style={styles.title}>
          {title}
        </Text>
        <Text>
          {date}
        </Text>
      </View>
      <View>
        <Button color={styles.button.color} title="View" onPress={props.onPress} />
      </View>
    </View>
  );
};

export default BloodTestCard;
