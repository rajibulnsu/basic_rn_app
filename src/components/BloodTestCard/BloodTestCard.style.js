import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  cardLayout: {
    flexDirection: 'row',
    alignItems: 'center',
    maxWidth: '100%',
    height: 90,
    margin: 10,
    padding: 20,
    borderRadius: 5,
    shadowColor: 'black',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.25,
    shadowRadius: 5,
    elevation: 5,
    backgroundColor: 'white',
  },
  text: {
    flex: 1,
  },
  title: {
    fontSize: 20,
  },
  button: {
    color: '#f5785a',
  },
});

export default styles;
