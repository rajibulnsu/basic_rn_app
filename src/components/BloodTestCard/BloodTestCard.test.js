import React from 'react';
import renderer from 'react-test-renderer';

import BloodTestCard from './BloodTestCard';

describe('BloodTestCard', () => {
  it('should match the snapshot', () => {
    const item = {
      title: 'title',
      date: '01.01.2019',
    };
    const bloodTestCard = renderer.create(<BloodTestCard item={item} onPress={() => {}}/>);

    expect(bloodTestCard).toMatchSnapshot();
  });
});
