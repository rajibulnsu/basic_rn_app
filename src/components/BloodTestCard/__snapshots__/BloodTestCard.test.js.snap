// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BloodTestCard should match the snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderRadius": 5,
      "elevation": 5,
      "flexDirection": "row",
      "height": 90,
      "margin": 10,
      "maxWidth": "100%",
      "padding": 20,
      "shadowColor": "black",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 5,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
        }
      }
    >
      title
    </Text>
    <Text>
      01.01.2019
    </Text>
  </View>
  <View>
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#f5785a",
              },
            ]
          }
        >
          View
        </Text>
      </View>
    </View>
  </View>
</View>
`;
